File: java/memento/TextEditor.java
package memento;

import java.util.Stack;

public class TextEditor {

    private String content;

    public TextEditor() {
        this.content = "";
    }

    public TextEditorState save() {
        return new TextEditorState(content);
    }

    public void restore(TextEditorState textEditorState) {
        content = textEditorState.getContent();
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}

File: typescript/memento/text-editor.ts
import { TextEditorState } from "./TextEditorState";

export class TextEditor {
  private content: string;
  constructor() {
    this.content = "";
  }

  public setContent(content: string) {
    this.content = content;
  }
  public getContent(): string {
    return this.content;
  }
  save() {
    return new TextEditorState(this.content);
  }
  restore(textEditorState: TextEditorState | null) {
    if (!textEditorState) this.content = "";
    else this.content = textEditorState.getContent();
  }
}
